File: go/response.go
package boohee

import "encoding/json"

// Response API 响应
//
// 包装统一的响应结构: {code, message, now, data}
type Response struct {
	Code    int             `json:"code"`
	Message string          `json:"message"`
	Now     int64           `json:"now"`
	Data    json.RawMessage `json:"data"`
}

// IsSuccess 判断请求是否成功（code == 0）
func (r *Response) IsSuccess() bool {
	return r.Code == 0
}

// Err 如果不成功则返回 APIError
func (r *Response) Err() error {
	if r.IsSuccess() {
		return nil
	}
	return &APIError{Code: r.Code, Message: r.Message}
}

// UnwrapData 将 data 反序列化到目标对象
func (r *Response) UnwrapData(v interface{}) error {
	if r.Data == nil {
		return nil
	}
	return json.Unmarshal(r.Data, v)
}

File: nodejs/src/response.ts
import { ApiError } from './errors';

/**
 * API 响应
 *
 * 包装统一的响应结构: {code, message, now, data}
 */
export class Response {
  readonly code: number;
  readonly message: string;
  readonly now: number;
  readonly data: Record<string, any> | null;

  constructor(raw: { code?: number; message?: string; now?: number; data?: Record<string, any> | null }) {
    this.code = raw.code ?? -1;
    this.message = raw.message ?? '';
    this.now = raw.now ?? 0;
    this.data = raw.data ?? null;
  }

  /** 判断请求是否成功（code == 0） */
  isSuccess(): boolean {
    return this.code === 0;
  }

  /** 如果请求失败则抛出 ApiError */
  raiseForError(): void {
    if (!this.isSuccess()) {
      throw new ApiError(this.code, this.message);
    }
  }

  /** 从 data 中获取字段 */
  get<T = any>(key: string, defaultValue?: T): T | undefined {
    if (this.data == null) return defaultValue;
    return (this.data[key] as T) ?? defaultValue;
  }

  toString(): string {
    return `Response(code=${this.code}, message=${this.message})`;
  }
}
